src/main/resources/init_db.sql

```diff
@@ -0,0 +1,5 @@
+CREATE TABLE IF NOT EXISTS books (
+    id BIGINT AUTO_INCREMENT PRIMARY KEY,
+    title VARCHAR(255) NOT NULL,
+    price BIGINT NOT NULL
```

Comment:
you're using `BigDecimal` in the code and at the same time `BIGINT` in the db

you should have used `DECIMAL`